### DISCUSSION TITLE
Exporting metrics in batches

### DISCUSSION THREAD

**Discussion by paul-ohl** [open-telemetry/opentelemetry-cpp#1849] [Q&A] [answered]
Hi everybody, I'm pretty new to opentelemetry, I hope this is the appropriate place to ask this.

---

**My situation:**

I need to monitor the evolution of a value every second, and would like to send the collected data every minute, but I need to have the details, I can't do any aggregation on the data.

**What I've already tried:**

I have found [this section](https://opentelemetry.io/docs/reference/specification/metrics/sdk/#exportbatch) of the documentation that seems to be exactly what I'm looking for, but I can't find any examples using it, and when I looked in the code, it seems to be declared in [push_metric_exporter.h](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/include/opentelemetry/sdk/metrics/push_metric_exporter.h#L35) but not implemented anywhere (other than in files inside the test directory). 

Also I haven't found any open issues that seem to be similar to mine.

My current guess is that the function is not implemented yet, or maybe just not for metrics, but it would be super helpful to know for sure.

Thank you very much for your attention!

**Comment by lalitb** [comment]
Help me understand your use case better  - you application is generating the measurement every second, and you want them to be exporter every minute. If that is the case, you can use the periodic metric reader which reads the metric data every minute. You application can either invoke Instrument::Record() every second, or create a callback with async instrument to be invoked every minute.

**Reply by paul-ohl** [reply]
Yes, exactly what you describe, but when I use the Record method of the Histogram it doesn't give me the detail of when each value was recorded.

I guessed that you were talking about the Histogram because I haven't found other classes that have this method.

**Comment by lalitb** [comment, marked as answer]
So you want to export the non-aggregated measurements every minute  - this is not supported by OpenTelemetry specs.  Each recorded measurement need to go through some aggregation. There are some discussions around that - https://github.com/open-telemetry/opentelemetry-specification/issues/617.

**Reply by paul-ohl** [reply]
Ok no problem, our app should still work with data being sent every second, or with an aggregation. But it is good to know that it is not supported, as it will save me quite a bit of time researching it!

Thank you so much for your help, have a nice day!
